apps/src/blockly/addons/functionEditor.js

```diff
@@ -370,4 +372,33 @@ export default class FunctionEditor {
     this.editorWorkspace.clear();
     Blockly.Events.enable();
   }
+
+  // If the definition block has a description property
+  // under definitionBlock.description, move it to the description field
+  // of the block, and delete the description property from the definition block.
+  // If the block does not have a description field, don't move the description.
+  moveDescriptionToBlockField(definitionBlock) {
+    if (
+      definitionBlock.description &&
+      definitionBlock.inputList &&
+      definitionBlock.inputList[1]?.fieldRow &&
+      definitionBlock.inputList[1].fieldRow[1]
+    ) {
+      definitionBlock.inputList[1].fieldRow[1].setValue(
+        definitionBlock.description
+      );
+      definitionBlock.description = undefined;
+      // Also update the description of the hidden definition block.
+      // Events don't run on custom properties such as description.
+      const topBlocks = Blockly.getHiddenDefinitionWorkspace().getTopBlocks();
+      const blockToUpdate = topBlocks.find(
+        topBlock =>
+          topBlock.getProcedureModel().getId() ===
+          this.block.getProcedureModel().getId()
```

Comment:
You're mixing the use of `definitionBlock` and `this.block` in this method, but `definitionBlock` and `this.block` are the same, right? Can/should we standardize on one or the other?

Reply:
good catch, I moved this logic around a bit and did not clock that I was just passing in `this.block` now. I'll update to use `this.block`